packages/test/test-end-to-end-tests/src/test/waitForSummary.spec.ts

```diff
@@ -0,0 +1,220 @@
+/*!
+ * Copyright (c) Microsoft Corporation and contributors. All rights reserved.
+ * Licensed under the MIT License.
+ */
+
+import { strict as assert } from "assert";
+
+import { bufferToString, stringToBuffer } from "@fluid-internal/client-utils";
+import { describeCompat } from "@fluid-private/test-version-utils";
+import {
+	LoaderHeader,
+	type IContainer,
+	type IHostLoader,
+} from "@fluidframework/container-definitions/internal";
+import type { IContainerExperimental } from "@fluidframework/container-loader/internal";
+import type {
+	ConfigTypes,
+	IConfigProviderBase,
+	IRequestHeader,
+} from "@fluidframework/core-interfaces";
+import type { ISharedMap } from "@fluidframework/map/internal";
+import {
+	type ITestObjectProvider,
+	type ITestContainerConfig,
+	createSummarizer,
+	summarizeNow,
+	type ChannelFactoryRegistry,
+	createAndAttachContainer,
+	DataObjectFactoryType,
+	type ITestFluidObject,
+	waitForContainerConnection,
+	timeoutAwait,
+} from "@fluidframework/test-utils/internal";
+
+// eslint-disable-next-line import/no-internal-modules
+import { loadOffline, getPendingOps } from "./offline/offlineTestsUtils.js";
+
+const waitForSummary = async (
+	provider: ITestObjectProvider,
+	container: IContainer,
+	testContainerConfig: ITestContainerConfig,
+	summaryVersion?: string,
+) => {
+	const { summarizer, container: summarizingContainer } = await createSummarizer(
+		provider,
+		container,
+		testContainerConfig,
+		summaryVersion,
+	);
+	await provider.ensureSynchronized();
+	const result = await summarizeNow(summarizer);
+	summarizingContainer.close();
+	return result.summaryVersion;
+};
+
+const configProvider = (settings: Record<string, ConfigTypes>): IConfigProviderBase => ({
+	getRawConfig: (name: string): ConfigTypes => settings[name],
+});
+
+describeCompat(
+	"Offline tests that wait for a summary",
+	"NoCompat",
+	(getTestObjectProvider, apis) => {
+		const mapId = "map";
+		const { SharedMap } = apis.dds;
+		const registry: ChannelFactoryRegistry = [[mapId, SharedMap.getFactory()]];
```

Comment:
Do you really need a map? It would be simpler and less code to use the default `root` DDS that the test data object exposes.

Reply:
Map is not really needed but I'm trying to keep tests as close as we have them as stashedOps.spec.ts, so it's more just for consistency purposes.